packages/truffle-compile/profiler/index.js

```diff
@@ -0,0 +1,266 @@
+// Compares .sol files to their .sol.js counterparts,
+// determines which .sol files have been updated.
+
+const path = require("path");
+const async = require("async");
+const CompilerSupplier = require("../compilerSupplier");
+const expect = require("truffle-expect");
+const findContracts = require("truffle-contract-sources");
+const semver = require("semver");
+const debug = require("debug")("compile:profiler");
+const { readAndParseArtifactFiles } = require("./readAndParseArtifactFiles");
+const {
+  minimumUpdatedTimePerSource
+} = require("./minimumUpdatedTimePerSource");
+const { findUpdatedFiles } = require("./findUpdatedFiles");
+const { isExplicitlyRelative } = require("./isExplicitlyRelative");
+const { getImports } = require("./getImports");
+
+module.exports = {
+  updated(options, callback) {
```

Comment:
What do you think of making this callback-style function be a wrapper around an async function? To get rid of the `.then()`s (and same with `required_sources`)

Reply:
Ultimately I would just like to make it a straight-up `async` function but I guess we should have a conversation about breaking changes and Truffle in general. Otherwise I suppose making it wrap an `async` function seems like a fine idea.

Reply:
Up to you if you want the dual interface. I was just saying the wrapper is sufficient and will help make the code nicer to look at.